### PR TITLE
[redesign] layout for editor components

### PR DESCRIPTION
**Pull request by thomasheyenbrock**
Part of #2325.

This implements the new design for the layout for the editor components:
- Writable editors with white background (query, variables, headers)
- Read-only response editor with gray background
- Toggle button for the editor tools (variable & header editors)

### CODE REVIEW

**Review by JoviDeCroock — APPROVED** [review]

File: packages/graphiql/src/components/GraphiQL.tsx [hunk 1]
```diff
@@ -745,109 +748,134 @@ class GraphiQLWithContext extends React.Component<
...
-                          Request Headers
-                        </div>
-                      )}
+                          Variables
+                        </UnStyledButton>
+                        {this.props.headerEditorEnabled ? (
+                          <UnStyledButton
+                            className={
+                              this.state.activeSecondaryEditor === 'header'
+                                ? ' active'
```
Comment (B2o5T) [inline packages/graphiql/src/components/GraphiQL.tsx:810]:
```suggestion
                                ? 'active'
```

Comment (thomasheyenbrock) [reply packages/graphiql/src/components/GraphiQL.tsx:810]: Good catch, same for the button for variables (the other class name there was unused) 👍

**Review by acao — APPROVED** [review]
beautiful! `codemirror` and the tab UI for variables/headers looks wonderful with the light. and I see the other editors are all looking nice. The redesign is emerging!
Screenshot 2022-06-27 at 21 28 30

also, I see you deleting css files there! so exciting to see this in motion

**Comment by acao** [comment]
I gasped aloud! so spectacular! How will `monaco-editor` compete? 😆  Jk. this is going to be amazing for docs sites!

Screenshot 2022-06-27 at 21 34 45

**Review by acao — APPROVED** [review]

File: packages/graphiql-react/src/editor/style/codemirror.css [hunk 2]
```diff
@@ -34,6 +34,9 @@
  */
 
 .cm-s-graphiql {
+  /* Default to punctuation */
+  color: var(--color-neutral-60);
```
Comment (acao) [inline packages/graphiql-react/src/editor/style/codemirror.css:38]: how does this work when other codemirror themes are introduced? Would it be easy for them to override this using css variables to suit a different theme if they didn't want to impact the rest of the styling? 

for example, a light graphiql theme with a dark codemirror theme, or vice versa?

**Pull request merged**